### PR TITLE
Avoid misleading error when Sugar connexion is not performed successfully

### LINKED ISSUE #16: UnboundLocalError: local variable 'conn' referenced before assignment
Sentry Issue: [BACKOFFICE-1B7](https://sentry.io/organizations/fever/issues/2624393654/?referrer=github_integration)

```
JSONDecodeError: Expecting value: line 1 column 1 (char 0)
(1 additional frame(s) were not displayed)
...
  File "pysugarcrm/pysugarcrm.py", line 56, in __init__
    response = requests.post(login_url, data=json.dumps(data)).json()
  File "requests/models.py", line 900, in json
    return complexjson.loads(self.text, **kwargs)
  File "__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None

UnboundLocalError: local variable 'conn' referenced before assignment
(21 additional frame(s) were not displayed)
...
  File "django/utils/functional.py", line 35, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "utils/decorators.py", line 129, in wrapper
    result = fn(*args, **kwargs)
  File "core/domain/partner/partner.py", line 344, in contracts_crm
    platform="fever-{}".format(uuid.uuid4().hex),
  File "contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "pysugarcrm/pysugarcrm.py", line 153, in sugar_api
    conn.close()
```

### PR DESCRIPTION
**Pull request by gprossignoli**
[SD-1272](https://feverup.atlassian.net/browse/SD-1272)
Related issue: #16 

If trying to connect to SugarCRM ends with a exception. The finally statement, will try to access a non-existing variable (conn). This PR changes this.

### How to test:
From fever2 BO, try to create a plan with a invalid Partner/Contract ID. you should get this error:
image

### CODE REVIEW

File: pysugarcrm/pysugarcrm.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def __init__(
         """
         Updates secret token to start making requests
         """
-
+        raise APIException()
```
Comment (miguelgf) [inline pysugarcrm/pysugarcrm.py]: not sure this is a good idea

Comment (gprossignoli) [reply pysugarcrm/pysugarcrm.py]: bad idea yes, sorry

**Review by lpl — APPROVED** [review]
Works fine!

Also the error when conn var is null is no longer present:

image

**Comment by lpl** [comment]
Let's add a test that covers the fix and update the HISTORY.rst, AUTHORS.rst and setup.py files to prepare this for the next release.

File: tests/test_pysugarcrm.py [hunk 2]
```diff
@@ -59,3 +59,14 @@ def test_context_manager(self):
...
+    @responses.activate
+    def test_context_manager_without_access_token(self):
+        responses.add(
+            responses.POST,
+            'http://testserver.com/rest/v10/oauth2/token',
+            body=json.dumps({}),
+            status=200
+        )
+
+        self.assertRaises(ValueError, func=sugar_api('http://testserver.com/', "admin", "12345"))
```
Comment (lpl) [inline tests/test_pysugarcrm.py]:
```suggestion
        with self.assertRaises(ValueError):
            sugar_api('http://testserver.com/', "admin", "12345")
```

I find something like this easier to read and understand, what do you think?

Comment (gprossignoli) [reply tests/test_pysugarcrm.py]: looks better for me too.

**Pull request merged**